docs/releases/0.8.0.rst

```diff
@@ -109,7 +109,7 @@ Breaking Changes
    (`fe453c3 <https://github.com/dhermes/bezier/commit/fe453c3839b19ce4a85dfd0b5ad78f71a0973daf>`__).
    Now when doing geometric curve-curve intersection, parallel linearized
    segments are handled by checking if the convex hulls collide and then
-   (if they do) using a modifed Newton iteration to converge to a root.
+   (if they do) using a modified Newton iteration to converge to a root.
```

Comment:
Unfortunately releases are "frozen in time" so e.g. https://bezier.readthedocs.io/en/0.8.0/releases/latest.html will "never" see this fix.

(In theory the page could be updated by "moving" the tag somehow but it would require a lot of effort for minimal returns.)

Getting the typo fix backported to https://github.com/dhermes/bezier/releases/tag/0.8.0 is easy though